### PR TITLE
ebpf: remove interpreter map

### PR DESCRIPTION
**Pull request by yanivagman**
interpreter_map is used to save interpreter file info per process. Now that we have the proc_info_map, we can save it there and remove the interpreter map.

### CODE REVIEW

File: pkg/ebpf/c/tracee.bpf.c [hunk 1]
```diff
@@ -6762,20 +6753,29 @@ int BPF_KPROBE(trace_load_elf_phdrs)
     if (!should_trace((&data)))
         return 0;
 
-    file_info_t elf = {};
+    proc_info_t *proc_info = bpf_map_lookup_elem(&proc_info_map, &data.context.task.host_pid);
+    if (proc_info == NULL) {
```
Comment (NDStrahilevitz) [inline pkg/ebpf/c/tracee.bpf.c:6757]: we can use `unlikely` here considering the later comment.

File: pkg/ebpf/c/tracee.bpf.c [hunk 2]
```diff
@@ -3828,9 +3821,7 @@ int tracepoint__sched__sched_process_exit(struct bpf_raw_tracepoint_args *ctx)
     // evaluate should_trace before removing this pid from the maps
     bool traced = should_trace(&data);
 
-    // Remove this pid from all maps
     bpf_map_delete_elem(&task_info_map, &data.context.task.host_tid);
```
Comment (NDStrahilevitz) [inline pkg/ebpf/c/tracee.bpf.c:3824]: Just making sure, this stays here and shouldn't move to the `sched_process_free` program?

**Review by NDStrahilevitz — APPROVED** [review]
LGTM, needs a rebase anyway so I left a suggestion and a question.

File: pkg/ebpf/c/tracee.bpf.c (hunk 2 above)
Comment (yanivagman) [reply pkg/ebpf/c/tracee.bpf.c:3824]: Yes, this should stay here since it is tid and not pid

**Pull request merged**